packages/cache/ARCHITECTURE.md

```diff
@@ -0,0 +1,56 @@
+# TODO: title
+
+## Assumptions
+
+- The `CacheRegistry` will be generated by the "Codegen" system where the full
+  set of query keys are known ahead of time. In Athena's case that will mean
+  that inidivual query ids like `query-id-123` will map to **exactly** the return type of that query.
```

Comment:
Probably more of an implementation detail question, Once we have the registry, in our use case world we will need to traverse this to start populating this in our cache records by entityUrns (the id in our usecase)? Do we need to explicitly store the quertids somewhere or we will get that info from the cache valye metadata?

Reply:
The cache does not need to do so.

The problem of tracking build-time metadata like query ids is for the codegen & runtime system (what we're tentatively calling Athena).

From the cache's point of view, this is something "user" supplied (although "user" here is other infrastructure).